### PR TITLE
Fix Get-Credential to not prompt twice when no parameter is specified

### LINKED ISSUE #2306: Get-Credential with no parameters prompts for user/password twice
Steps to reproduce
------------------
`Get-Credential`

Expected behavior
-----------------
```
PS> Get-Credential

Windows PowerShell credential request.
User: bar
Password for user bar: ********

UserName                     Password
--------                     --------
bar      System.Security.SecureString
```

Actual behavior
---------------
```
PS> Get-Credential

cmdlet Get-Credential at command pipeline position 1
Supply values for the following parameters:
Credential
User: foo
Password for user foo: ***


Windows PowerShell credential request.
User: bar
Password for user bar: ********


UserName                     Password
--------                     --------
bar      System.Security.SecureString
```

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSRemotingProtocolVersion      2.3
CLRVersion
BuildVersion                   3.0.0.0
WSManStackVersion              3.0
PSVersion                      6.0.0-alpha
PSEdition                      Core
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
SerializationVersion           1.1.0.1
GitCommitId                    v6.0.0-alpha.10-10-g87f6e8d04086da21093243acf9c6e11998e02ac8-dirty
```

### PR DESCRIPTION
**Pull request by daxian-dbw**
Fix #2306

### CODE REVIEW

**Comment by daxian-dbw** [comment]
@lzybkr can you please review this PR? Thanks!

File: test/powershell/Modules/Microsoft.PowerShell.Security/GetCredential.Tests.ps1 [hunk 1]
```diff
@@ -53,4 +57,15 @@ Describe "Get-Credential Test" -tag "CI" {
...
+    It "Get-Credential without parameters" {
+        $cred = $ps.AddScript("Get-Credential").Invoke() | Select-Object -First 1
+        $cred.gettype().FullName | Should Be "System.Management.Automation.PSCredential"
+        $netcred = $cred.GetNetworkCredential()
+        $netcred.UserName | Should be "John"
+        $netcred.Password | Should be "credTest"
+        ## Prompt should be "cmdlet Get-Credential at command pipeline position 1:Supply values for the following parameters::Credential"
+        ## But we should avoid comparing the whole string in case it's not running in en-us culture
+        $th.ui.Streams.Prompt[-1] | Should Match "Get-Credential"
+        $th.ui.Streams.Prompt[-1] | Should Match ":Credential"
```
Comment (lzybkr) [inline test/powershell/Modules/Microsoft.PowerShell.Security/GetCredential.Tests.ps1]: You could merge this with the previous `Match`.

**Review by lzybkr — COMMENTED** [review]
You know it's weird that we rely on the parameter binder to do the prompting if no arguments are passed, but the prompting is done here if there are arguments.

For example, I get the extra prompting:

```
cmdlet Get-Credential at command pipeline position 1
Supply values for the following parameters:
Credential
```

If I don't specify parameters. This isn't the best user experience, but as it is, I'd recommend always using `-Message` to avoid that extra text.

**Comment by daxian-dbw** [comment]
The `Credential` parameter set is the default parameter set, and `-Credential` is the only mandatory parameter in that parameter set, so it's natural that the parameter binder will kick in prompting for the missing mandatory parameter `-Credential` when no argument is specified.

**Comment by lzybkr** [comment]
Of course - but I'm asking a different question - can we get the same effect of prompting in `Get-Credential` without relying on the parameter binder?

For example, if we changed the default message/title in `Get-Credential`, we'd also want a similar change in the parameter binder. I don't think that's obvious or ideal.

But really - I'm just saying we should be seeing the parameter binding prompts when asking the user for a credential. Ever.

**Comment by daxian-dbw** [comment]
Got @lzybkr 's point now. The current change gives inconsistent prompting experience for `Get-Credential` and `Get-Credential -UserName bar`. Note that the default caption and message are not displayed for `Get-Credential`
```
PS D:\> Get-Credential

cmdlet Get-Credential at command pipeline position 1
Supply values for the following parameters:
Credential
User:
PS D:\>
PS D:\> Get-Credential -UserName bar

Windows PowerShell credential request.
Password for user bar:
```
This is because for `Get-Credential`, the prompting comes from the parameter binder, while the other comes from `Host.UI.PromptForCredential`. I will update the PR to provide consistent prompting experience.

**Comment by iSazonov** [comment]
@daxian-dbw Please take a look #2327

**Comment by McAndersDK** [comment]
I approve this fix 👍 

But why not move the last writeobject to the try ?
```
try
{
    Credential = this.Host.UI.PromptForCredential(_title, _message, _userName, string.Empty);
    WriteObject(Credential);
}
```

**Comment by daxian-dbw** [comment]
@McAndersDK it's to avoid a very very rare edge case -- `WriteObject` somehow throws `ArgumentException`. If that really happens, we should allow that exception to propagate and be handled by the engine.

**Review by lzybkr — APPROVED** [review]

**Pull request merged**